components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -787,6 +802,131 @@ void BraveProfileSyncServiceImpl::LoadSyncEntityInfo(
   }
 }
 
+bool BraveProfileSyncServiceImpl::IsOtherBookmarksFolder(
+    const jslib::SyncRecord* record) const {
+  auto bookmark = record->GetBookmark();
+  if (!bookmark.isFolder)
+    return false;
+
+  std::string other_node_object_id;
+  if (model_->other_node()->GetMetaInfo("object_id", &other_node_object_id) &&
+      record->objectId == other_node_object_id)
+    return true;
+
+  if (bookmark.order == tools::kOtherNodeOrder &&
+      bookmark.site.title == tools::GetOtherNodeName() &&
+      bookmark.site.customTitle == tools::GetOtherNodeName()) {
+    return true;
+  }
+
+  return false;
+}
+
+void BraveProfileSyncServiceImpl::ProcessOtherBookmarksFolder(
+    const jslib::SyncRecord* record,
+    bool* pass_to_syncer) {
+  std::string other_node_object_id;
+    // Save object_id for late joined desktop to catch up with current id
+    // iteration
+  if (!model_->other_node()->GetMetaInfo("object_id", &other_node_object_id) &&
+      record->action == jslib::SyncRecord::Action::A_CREATE) {
+    tools::AsMutable(model_->other_node())->SetMetaInfo("object_id",
+                                                        record->objectId);
+  } else {
+    // If late joined desktop has bookmarks in other_node before joining sync
+    // chain
+    if (other_node_object_id != record->objectId &&
+        other_node_object_id ==
+          tools::GenerateObjectIdForOtherNode(std::string())) {
+      tools::AsMutable(model_->other_node())->SetMetaInfo("object_id",
+                                                          record->objectId);
+    }
+    // DELETE won't reach here, because [DELETE, null] => [] in
+    // resolve-sync-objects but children records will go through. And we don't
+    // need to regenerate new object id for it.
+
+    // Handle MOVE, RENAME
+    // REORDER (move under same parent) will be ignored
+    // Update will be resolved as Create because [UPDATE, null] => [CREATE]
+    auto bookmark = record->GetBookmark();
+    if ((bookmark.order != tools::kOtherNodeOrder &&
+         !bookmark.parentFolderObjectId.empty()) ||
+        bookmark.site.title != tools::GetOtherNodeName() ||
```

Comment:
This line means on remote device which created the `record` , `Other bookmarks`node was renamed.
So this condition will also be triggered if current device will change own device language between browser restart.

It is interesting thing, what will happened, if we will take two desktop browsers on a different computers, each of that computers will have different system interface language. Will they send update to each to other endlessly?
Going to check this: desktop [EN], desktop [UA], Android [no matter]

Reply:
I think there is an issue here, but not exactly what @AlexeyBarabash is saying. I think we always need to save the folder name to sync in English

Reply:
fixed in 07e019e5ed6b0dc01072bcc11f5b9bfa60ef9150